### PR TITLE
Encryption

### PR DESCRIPTION
**Pull request by tonsV2**
The goal of this PR is to implement encryption of parameters for deployment instances.

To avoid further code duplication this PR will disable encryption for the regular (non-deployment) instances.

### CODE REVIEW

File: pkg/instance/handler.go [hunk 1]
```diff
@@ -65,15 +65,15 @@ func (h Handler) DeployDeployment(c *gin.Context) {
...
 
-	deployment, err := h.instanceService.FindDeploymentById(id)
+	deployment, err := h.instanceService.FindDecryptedDeploymentById(id)
 	if err != nil {
 		_ = c.Error(err)
 		return
 	}
 
-	canRead := handler.CanReadDeployment(user, deployment)
+	canRead := handler.CanWriteDeployment(user, deployment)
```
Comment (radnov) [inline pkg/instance/handler.go:74]: Should the variable here be `canWrite`?

File: pkg/instance/repository.go [hunk 2]
```diff
@@ -75,8 +75,61 @@ func (r repository) FindDeploymentById(id uint) (*model.Deployment, error) {
...
 
+func (r repository) FindDeploymentInstanceById(id uint) (*model.DeploymentInstance, error) {
+	var instance *model.DeploymentInstance
+	err := r.db.
+		Preload("GormParameters").
+		First(&instance, id).Error
+
+	if err != nil {
+		if errors.Is(err, gorm.ErrRecordNotFound) {
+			return nil, errdef.NewNotFound("instance not found by id: %d", id)
```
Comment (radnov) [inline pkg/instance/repository.go:86]: Should we also use the standard `"failed to ..."` error message prefix here?

File: pkg/instance/repository.go [hunk 3]
```diff
@@ -132,16 +183,16 @@ func (r repository) Unlink(instance *model.Instance) error {
...
-		return err
-	}
-
-	err = r.db.Session(&gorm.Session{FullSaveAssociations: true}).Save(instance).Error
+	/*
+		err := encryptParameters(key, instance)
+		if err != nil {
+			return err
+		}
+	*/
```
Comment (radnov) [inline pkg/instance/repository.go:194]: Are the commented out lines here intentional?

**Review by radnov — COMMENTED** [review]
Added a few comments 👍 

Note that I have not tested the changes.

File: pkg/instance/repository.go (hunk 3 above)
Comment (tonsV2) [reply pkg/instance/repository.go:194]: The entire method is to be deleted

File: pkg/instance/repository.go (hunk 2 above)
Comment (radnov) [reply pkg/instance/repository.go:86]: Yeah, there's a mix of `X not found ...` and `failed to ...` error messages in both new and old code. Only the latter is mentioned in [the style docs here](https://github.com/dhis2-sre/im-doc/tree/master/dev#errors), so we should either stick to it (and even go back and apply it across all error messages) or update the doc. 👀

File: pkg/instance/repository.go (hunk 3 above)
Comment (radnov) [reply pkg/instance/repository.go:194]: If it should be merged like this I'd add a `// TODO ...` above those lines.

**Review by radnov — APPROVED** [review]

**Pull request merged**